### DISCUSSION TITLE
Receiving and returning audio & video files using FileResponse

### DISCUSSION THREAD

**Discussion by masus04** [fastapi/fastapi#6284] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
tmp_file_dir = "/tmp/example-files"
Path(tmp_file_dir).mkdir(parents=True, exist_ok=True)


@media_router.post(
    path="/api/media-file",
    response_class=FileResponse,
)
async def post_media_file(file: UploadFile):
    """
    Receive File, store to disk & return it
    """
    # Write file to disk. This simulates some business logic that results in a file sotred on disk
    with open(os.path.join(tmp_file_dir, file.filename), 'wb') as disk_file:
        file_bytes = await file.read()

        disk_file.write(file_bytes)

        print(f"Received file named {file.filename} containing {len(file_bytes)} bytes. ")

        return FileResponse(disk_file.name, media_type=file.content_type)
```

### Description

The example code I added accepts any file a user uploads, stores it to disk and returns it using a FileResponse.

This approach works nicely for images, however when uploading audio or video, the result cannot be displayed in the /docs properly and I assume the same would be the case for any consumer of the API.

I assume this is merely a matter of adding some metadata either to the file or the FileResponse but I was unable to find out what or how to add them.

### Operating System

Linux

### FastAPI Version

0.78.0

### Python Version

3.10.4

**Comment by iudeen** [comment]
What you have done is correct... not sure if SwaggerUI supports audio playback.

**Comment by masus04** [comment]
@iudeen Thanks for the quick reply.

OpenAPI/Swagger does seem to support audio. At least it shows an audio player, but the controls are all disabled, indicating to me that it probably can't open the received data.

I also tested the API using PostMan: Both audio and video were playing but none of them had any sound.

When I downloaded the response from postman and played it using VLC media player, they worked fine. I'm assuming it has to do with encoding..

**Comment by iudeen** [comment]
This code works. Swagger doc does a `GET` request. So you need to have your route as `GET` method. 

```python
from fastapi import FastAPI
from fastapi.responses import FileResponse

app = FastAPI()


@app.get(
    path="/api/media-file"
)
async def post_media_file():
    return FileResponse("audio.mp3", media_type="audio/mpeg")
```

**Comment by luebke-dev** [comment, marked as answer]
works fine with curl:
`curl --output test.mp3 -F file=@ch.mp3 http://localhost:8000/api/media-file`

this returns a valid mp3 file with audio

the problem here is the swagger ui, when you upload a file through swagger the player tries to make a GET request to the same path.

You see that in the fastapi log:
```
INFO:     127.0.0.1:34792 - "POST /api/media-file HTTP/1.1" 200 OK
INFO:     127.0.0.1:34792 - "GET /api/media-file HTTP/1.1" 405 Method Not Allowed
```

You can also see that after upload the following audio element is attached to the DOM:
```html
<audio controls>
<source src="http://0.0.0.0:8000/api/media-file" type="audio/mpeg">
</audio>
```
Since the player is attached to the DOM after the request with the URL as SRC it tries to make GET request to that url and this GET request obviously fails because your route is a POST route.

**Comment by masus04** [comment]
Thank you very much for the explanation. Since this seems to be an issue with OpenAPI/Swagger I don't suppose there's anything that can be done on the FastAPI side, I'll close the issue.

**Comment by luebke-dev** [comment]
If you really need to listen to the audio file in the swagger ui you couldt try a html response where you return an audio element wirh the base64 encoded audio file as src. But im not sure if it will be rendered

**Comment by EmreOzkose** [comment]
Hi @masus04 ,

I have same issue, did you solve that ?

**Comment by ChrisDeadman** [comment]
Just found this issue after wasting 2 hours wondering why that play button is grayed out. I was convinced I must have done something wrong in my endpoint even if it works without issues with curl.
This really is a bit of a bummer because it renders swagger unusable for text-to-speech applications.
